core/templates/pages/preferences-page/preferences-page.component.ts

```diff
@@ -87,6 +91,24 @@ angular.module('oppia').component('preferencesPage', {
         _saveDataItem('user_bio', userBio);
       };
 
+      ctrl.checkBioChanged = function() {
+        ctrl.allowreload = false;
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
+
+      ctrl.confirmBeforeLeaving = function(e) {
+        if (!ctrl.allowreload) {
+          // This message is irrelevant, but is needed to trigger the
+          // confirmation before leaving.
```

Comment:
Not sure what does this mean, what message?

Reply:
I'm guessing `e.returnValue = 'Sure?';` is the message here. Basically the 'Sure?' part

Reply:
```suggestion
          // The exact value in returnValue is not relevant, but it needs to be set
          // in order to trigger a confirmation before leaving.
```